### DISCUSSION TITLE
Handling exceptions raised by `forward` during a training step

### DISCUSSION THREAD

**Discussion by jangop** [Lightning-AI/pytorch-lightning#15188] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
A model might raise an exception during its `forward`, as, for example, torchvision's current `GeneralizedRCNN` does:

```python
    if degenerate_boxes.any():
        ...
        torch._assert(False,...)
```

https://github.com/pytorch/vision/blob/0610b13ac4af3717f538454a9c6b1f441cb386f3/torchvision/models/detection/generalized_rcnn.py#L95

While one might consider this problematic in general, I would like to get around it by essentially skipping a batch if an exception occurs:

```python
class OuterModel(pl.LightningModule):
    ...
        self.rcnn = torchvision...maskrcnn...
    ...
    def training_step(self, batch, batch_idx):
        try:
            loss_dict = self.rcnn(batch)  # In training mode, RCNN directly returns several losses
        except AssertionError:
            log.exception(...)
            return ???
    return sum(loss for loss in loss_dict.values())
```

The problem is that PL expects a loss returned. Is there any sensible way to go about this?

**Comment by akihironitta** [comment]
Returning `None` from training_step is supported in some cases. See https://pytorch-lightning.readthedocs.io/en/stable/common/lightning_module.html#training-step
